dask/dataframe/groupby.py

```diff
@@ -914,11 +934,25 @@ def _build_agg_args_simple(result_column, func, input_column, impl_pair):
     )
 
 
-def _build_agg_args_var(result_column, func, input_column):
+def _build_agg_args_var(result_column, func, func_args, func_kwargs, input_column):
     int_sum = _make_agg_id("sum", input_column)
     int_sum2 = _make_agg_id("sum2", input_column)
     int_count = _make_agg_id("count", input_column)
 
+    # we don't expect positional args here
+    if func_args:
+        raise TypeError(
+            f"aggregate function '{func}' got unexpected positional arguments {func_args}"
+        )
+
+    # and we only expect ddof=N in kwargs
+    expected_kwargs = {"ddof"}
+    unexpected_kwargs = set(func_kwargs.keys()) - expected_kwargs
+    for arg in unexpected_kwargs:
+        raise TypeError(
+            f"aggregate function '{func}' got an unexpected keyword argument '{arg}'"
+        )
```

Comment:
For multiple unexpected `kwargs`, it'd probably be a bit more informative to just let users see all the invalid `kwargs` at once. Otherwise, it may take some iteration to get rid of them all. 

```suggestion
    if unexpected_kwargs:
        raise TypeError(
            f"The aggregate function '{func}' supports {expected_kwargs} keyword arguments, but got {unexpected_kwargs}"
        )
```

Reply:
Note we'll need to make a corresponding change to `test_groupby_aggregate_partial_function_unexpected_kwargs` since we're updating the error message